internal/plugin/grpc_provider.go

```diff
@@ -142,13 +147,21 @@ func (p *GRPCProvider) GetProviderSchema() (resp providers.GetProviderSchemaResp
 		resp.ServerCapabilities.GetProviderSchemaOptional = protoResp.ServerCapabilities.GetProviderSchemaOptional
 	}
 
-	// set the global cache if we can
+	// Set the global provider cache so that future calls to this provider can use the cached value.
+	// Crucially, this doesn't look at GetProviderSchemaOptional, because the layers above could use this cache
+	// *without* creating an instance of this provider. And if there is no instance,
+	// then we don't need to set up anything (cause there is nothing to set up), so we need no call
+	// to the providers GetSchema rpc.
```

Comment:
```suggestion
        // Set the global provider cache so that future calls to this provider can use the cached value.
	// Crucially, this doesn't look at GetProviderSchemaOptional, because the provider was instantiated at this point already, and its schema can be stored to the cache for further use without the need to instantiate provider. 
```

Reply:
I think you're suggesting that we don't look at GetProviderSchemaOptional because for future calls this will now be instantiated and already called at least once.

But that's not the case, as this instance of the provider could be stopped (and that actually does indeed happen) right after this call finishes. So if we created a new instance of the provider, it would again require initialization.

Reply:
Ah, right. Thanks for clarification.